### DISCUSSION TITLE
Data Saving

### DISCUSSION THREAD

**Discussion by pietru2004** [bitbrain/pandora#105] [Q&A] [answered]
Based on https://github.com/bitbrain/pandora/issues/104
Is there way to get item ID for stuff like ores, herbs, wood and non unique items and then load item based on ID ?

**Comment by mrfelipemartins** [comment, marked as answer]
Assuming you have an instance of PandoraEntity you can do this to retrieve it's ID:

```gdscript
var id = iron_ore.get_entity_id()
```

And you can load an item from ID like this:

```gdscript
var iron_ore = Pandora.get_entity(id)
```

**Reply by pietru2004** [reply]
ok thanks for help

**Reply by pietru2004** [reply]
I guess I should be able to make dictionary and store items counts under keys made of entity ids

**Reply by mrfelipemartins** [reply]
Not sure if I understood what you need but if you want to store the amount of an Item a player has, a good approach would be having a int property named for example "amount" and update an instance of your item's amount property, assuming your item can be stacked.

**Reply by pietru2004** [reply]
I like to use system of array or dictionary, having key that is items ref and count of item, if item not unique
